src/main/java/lotto/Lottos.java

```diff
@@ -0,0 +1,37 @@
+package lotto;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Lottos {
+
+  private final List<Lotto> lottos;
+
+  public Lottos(List<Lotto> lottos) {
+    this.lottos = lottos;
+  }
+
+  public int count() {
+    return lottos.size();
+  }
+
+  public LottoMatchResult calculateMatchCounts(Lotto winningNumbers) {
+    Map<Integer, Integer> result = new HashMap<>();
+    for (Lotto lotto : lottos) {
+      int match = lotto.countMatchingNumbers(winningNumbers);
+      result.put(match, result.getOrDefault(match, 0) + 1);
+    }
+    return new LottoMatchResult(result);
```

Comment:
```suggestion
    LottoMatchResult result = new LottoMatchResult();
    for (Lotto lotto : lottos) {
      int match = lotto.countMatchingNumbers(winningNumbers);
      result.match(match);
    }
    return result;
```

위와 같이 Map에 대한 상태 변경을 Map을 상태 값으로 가지고 있는 객체가 담당하도록 상태 값을 가지는 도메인 객체를 좀 더 능동적인 객체로 바라보기 위해 노력해 본다.
이렇게 구현했을 때 테스트 관점에서 얻을 수 있는 잇점은?